tlspuffin/src/openssl/mod.rs

```diff
@@ -135,7 +130,23 @@ impl Put<TLSProtocolBehavior> for OpenSSL {
     }
 
     fn reset(&mut self, _agent_name: AgentName) -> Result<(), Error> {
-        bindings::clear(self.stream.ssl()); // FIXME: Add non-clear method like in wolfssl
+        if self.config.use_clear {
+            bindings::clear(self.stream.ssl());
+        } else {
+            self.stream = Self::new_stream(&self.ctx, &self.config).map_err(|err| {
+                Error::Put(format!("OpenSSL error during stream creation: {}", err))
+            })?;
+
+            // FIXME don't force-register a new claimer on reset
```

Comment:
It seems this modifies the behavior when using the OpenSSL vendor and when the `--put-use-clear` is not used. I think we should make `put-use-clear` the default behavior and makes it opt-out.

Reply:
The put-use-clear was required to reproduce a specific vulnerability which only works when using clear() instead of creating a new SSL context. 

We could discuss whether we want to remove this feature and bring that back eventually. There are likely a lot of similar options we want to have at some point that affect intitialization.

Reply:
Yeah this is kind of changing things, but in general, creating a new SSL struct is the right thing to do to keep things more isolated. So I would not be concerned here with changing the default.

This was more like a historic thing and changing this should not cause any problems.

Reply:
There are two dimensions here:
 1. efficiency: it is expected to be faster to clear versus to recreate SSL struct. I don't have figures measuring this though. Anyone has?
 2.  fuzzing capabilities: 
    - using CLEAR allows to reach bugs e.g., based on side-conditions from previous executions impacting current executions
    - not using CEAR could potentially allow to execute more traces, some of them triggering bugs, that would otherwise be rejected when using CLEAR.
    
I don't know how to balance this trade-off now but we don't have to before merging this PR I think.